### DISCUSSION TITLE
Parent route match in useMatchRoute

### DISCUSSION THREAD

**Discussion by morzel85** [TanStack/router#977] [Q&A] [answered]
Let's say I have a route like `/user/profile`, I can then use `useMatchRoute` to check if the current route is `/user/profile`.

I would also like to check if parent `/user` route is matched (for example to style parent navigation item). 

What's the best way to check if parent route is matching? Is it supported directly in Tanstack Router v1 or do I need to do some custom substring checks?

**Comment by schiller-manuel** [comment]
how would the parent route not match if the child route matches? I probably don't understand your problem, can you please create a minimal example on codesandbox?

**Reply by morzel85** [reply]
I need to verify if current URL is not only matching some nested route, but that it also matches the parent route.

I've modified the Basic example from the docs, assume that I'm interested to know if current '/posts/1' (nested route) is also matching the '/posts'. So I need to do something similar to what the Link component must be doing for its _active_ state detection. 

I'm looking for the idiomatic way (preferably avoiding custom string matching on current window location).

![image](https://github.com/TanStack/router/assets/7968512/9643e036-c020-48a1-bd95-f6b95df2aa74)

[Code sample](https://codesandbox.io/p/devbox/objective-darkness-4lcyjv?file=%2Fsrc%2Fmain.tsx%3A56%2C40&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrdn1gzu0007356ig0wqaaei%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrdn1gzu0002356ino1dcrxy%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrdn1gzu0004356isf40bjoo%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrdn1gzu0006356i1dwdfiei%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B53.96397209121525%252C46.03602790878475%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrdn1gzu0002356ino1dcrxy%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrdn1gzu0001356ilqm95vah%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clrdn739s00ay356ioc5qafqa%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A56%252C%2522startColumn%2522%253A40%252C%2522endLineNumber%2522%253A56%252C%2522endColumn%2522%253A40%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clrdn1gzu0002356ino1dcrxy%2522%252C%2522activeTabId%2522%253A%2522clrdn739s00ay356ioc5qafqa%2522%257D%252C%2522clrdn1gzu0006356i1dwdfiei%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrdn1gzu0005356iybrnjzv4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3001%252C%2522path%2522%253A%2522%252Fposts%252F1%2522%257D%255D%252C%2522id%2522%253A%2522clrdn1gzu0006356i1dwdfiei%2522%252C%2522activeTabId%2522%253A%2522clrdn1gzu0005356iybrnjzv4%2522%257D%252C%2522clrdn1gzu0004356isf40bjoo%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrdn1gzu0003356i4970pfo6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clrdn1gzu0004356isf40bjoo%2522%252C%2522activeTabId%2522%253A%2522clrdn1gzu0003356i4970pfo6%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by schiller-manuel** [reply, marked as answer]
You can use `matchRoute({to: '/posts', fuzzy: true})`.
There was a bug with fuzzy matching but this is fixed in v1.4.5 https://github.com/TanStack/router/releases/tag/v1.4.5

**Reply by morzel85** [reply]
Looks good in 1.4.5 with fuzzy, thanks for quick fix!

![image](https://github.com/TanStack/router/assets/7968512/fc00e5f5-e3b6-4ee7-b25a-6cbd5872305b)

**Comment by andyblack19** [comment]
@schiller-manuel Just building on this discussion as it's fairly related...
How am I able to perform the same fuzzy matching against a route **id**?

I want to style a parent element based on one of its child Links being active, but the child Links do not all have the same path prefix.

Example folder structure below (_resources.tsx has some beforeLoad auth checks in).
```
_resources/whitepapers.tsx
_resources/articles.tsx
```

This will render in a nav along the lines of:
```
Resources   <-- onclick expand/collapse the sub-menu. Need to style this element when any child link is active.
|-- Whitepapers
|-- Articles
```

I was hoping to be able to do something like: `matchRouteById({ to: "_resources", fuzzy: true })` ??

**Reply by schiller-manuel** [reply]
you could use https://tanstack.com/router/v1/docs/framework/react/api/router/useChildMatchesHook to find out if and which child of a particular route is matched
